crates/services/relayer/src/service/get_logs.rs

```diff
@@ -61,16 +61,22 @@ where
                         let page = page.reduce();
 
                         // Get the logs and return the reduced page.
-                        eth_node.get_logs(&filter).await.map(|logs| {
-                            Some((
-                                DownloadedLogs {
-                                    start_height: oldest_block,
-                                    last_height: latest_block,
-                                    logs,
-                                },
-                                page,
-                            ))
-                        })
+                        eth_node
+                            .get_logs(&filter)
+                            .await
+                            .inspect_err(|err| {
+                                tracing::error!(?err, "eth provider failed to get logs")
```

Comment:
This might be easier than trying to pipe the correct error through the stream unlike below.

Reply:
Not really following what you refer to here but feel free to update as I'm just about to log off for the night.